src/parser/core/tests/log-tools.js

```diff
@@ -0,0 +1,65 @@
+import EventEmitter from 'parser/core/modules/EventEmitter';
+import decompress from 'decompress';
+
+const _CACHE = {};
+
+// asynchronously load and parse a log. returns a promise that resolves
+// to the log object
+export function loadLog(key, searchPath='test-logs/') {
+  if(_CACHE[key] !== undefined) {
+    return Promise.resolve(_CACHE[key]);
+  }
+  return decompress(`${searchPath}${key}.zip`).then(files => {
+    const result = files.reduce((res, file) => {
+      res[file.path.split('.')[0]] = JSON.parse(file.data.toString());
+      return res;
+    }, {});
+
+    _CACHE[key] = result;
+    return result;
+  });
+}
+
+/**
+ * Suppress logging within the test.
+ *
+ * @param {boolean} log - Suppress console.log?
+ * @param {boolean} warn - Suppress console.warn?
+ * @param {boolean} error - Suppress console.error?
+ */
+export function suppressLogging(log, warn, error) {
+  const _console = {};
+  beforeEach(() => {
+    if(warn) {
+      _console.warn = console.warn;
+      console.warn = () => undefined;
+    }
+    if(log) {
+      _console.log = console.log;
+      console.log = () => undefined;
+    }
+    if(error) {
+      _console.error = console.error;
+      console.error = () => undefined;
+    }
+  });
+
+  afterEach(() => {
+    Object.keys(_console)
+      .forEach(key => { console[key] = _console[key]; });
+  });
+}
+
+export function parseLog(parserClass, log) {
+  const friendlies = log.report.friendlies.find(({id}) => id === log.meta.player.id);
+  const fight = log.report.fights.find(({id}) => id === log.meta.fight.id);
+  const parser = new parserClass(
+    log.report, 
+    friendlies,
+    fight,
+    log.combatants
+  );
+  parser.normalize(JSON.parse(JSON.stringify(log.events))).forEach(event => parser.getModule(EventEmitter).triggerEvent(event));
```

Comment:
This json encoding/parsing seems unnecessary.

Reply:
That's the simplest way to deep copy without libs. Normalizers sometimes mutate the event list